### ISSUE TITLE
UnreachableCode false positives when invoking an error function which cannot be resolved

### ISSUE THREAD

**Issue by dzirbel** [detekt/detekt#5111]
## Expected Behavior
When encountering a field or function named `error` which cannot be resolved by type resolution, the `UnreachableCode` rule shouldn't assume it is `kotlin.error()`.

## Observed Behavior
This unit test fails:
```
@Test
fun `does not report unreachable code after unknown function named error`() {
    val code = """
        import foo.error

        fun test(): Int {
            error("error")
            return 0
        }
    """
    assertThat(subject.compileAndLintWithContext(env, code)).isEmpty()
}
```

This one does not, since the function is not named `error`:
```
@Test
fun `does not report unreachable code after unknown function`() {
    val code = """
        import foo.bar
    
        fun test(): Int {
            bar("error")
            return 0
        }
    """
    assertThat(subject.compileAndLintWithContext(env, code)).isEmpty()
}
```

It also does not fail if `error` can be resolved (i.e. is declared in the code block itself), but does fail even if `error` is just a property and not even a function call.

## Context
This is a very esoteric false positive, which detekt probably can't solve directly since it relies on diagnostics provided by the Kotlin compiler for unreachable code detection. I found it for cases which use an Android view named `error` and are used in code via Kotlin synthetics, which can't be resolved at compile time by detekt. I can get around it easily enough, but mostly wanted to report for documentation purposes :)

## Your Environment
* Version of detekt used: 1.21.0

**Comment by BraisGabin** [comment]
Wow, I remember this issue and I thought I answered to it. I think that there is really little we can do here. I think that the only way to fix this is to use the compiler plugin so we have access to all those types that we are missing right now.

The "good news" is that kotlin `1.8` will remove the support for android-kotlin synthetics so that will reduce this issues.

I'm closing this because there is nothing we can do here. Anyway, thanks for reporting :)
